test/Microsoft.AspNetCore.Razor.Runtime.Test/TagHelpers/TagHelperOutputTest.cs

```diff
@@ -990,6 +992,34 @@ public void WriteTo_WritesFormattedTagHelper(TagHelperOutput output, string expe
             Assert.Equal(expected, writer.ToString(), StringComparer.Ordinal);
         }
 
+        // This tests a separate code path that's used by THO when the writer is an HtmlTextWriter.
+        // The output should be the same, but we do some specific perf optimizations on this path.
+        [Theory]
+        [MemberData(nameof(WriteTagHelper_InputData))]
+        public void WriteTo_WritesFormattedTagHelper_HtmlTextWriter(TagHelperOutput output, string expected)
```

Comment:
I'll update this comment. This feels like a good integration test to have even though it's not a separate code path anymore.